bazel/version/gen_tmpl.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+
+"""Generate a file from a template, substituting the provided key/value pairs.
+
+The file format should match Python's `string.Template` substitution rules:
+- `$$` for a literal `$`
+- `$identifier` for some key `identifier` to be substituted
+- `${identifier}` when adjacent text would be interpreted as part of the
+  identifier.
+
+The keys must be strings that are valid identifiers: `[_A-Za-z][_A-Za-z0-9]*`
+
+The values may not contain newlines or any vertical whitespace.
+
+The initial key/value pairs are read from the command line using repeated
+`--substitute=KEY=DEFAULT-VALUE` flags.
+
+Updated values for those keys will be read from any files provided to the
+`--status-file` flag. This flag can be given multiple times and the values will
+be read and updated from the files in order, meaning the last file's value will
+win. New keys are never read from these files. The file format parsed is chose
+to be compatible with Bazel's status file format, but any files with lines
+matching the regex `^(?P<key>[_A-Za-z][_A-Za-z0-9]) (?P<value>.*)$` may be used.
```

Comment:
This seems to be different from VAR_RE, would it be worth just pointing to that?

Reply:
Went with your suggestion to remove the regex, and so tried to just spell out the format here.